### ISSUE TITLE
External ipykernel break the plugins

### ISSUE THREAD

**Issue by impact27** [spyder-ide/spyder#15921]
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
When switching to an external ipykernel, the plugins do not update correctly.

For example, the variable explorer would show the namespace of the previous console instead of the current one

I know that the variable explorer  is not expected to work for ipykernels, but it should not show the content of another console.

### What steps reproduce the problem?

1. In a terminal: `python -m ipykernel`
2. Open spyder with one console named "Console 1"
3. Console > Connect to existing console > paste the kernel number (`kernel-XXXX.json`)
4. A new console (Console 2) opens
5. Go to console one, type `a = 1`
6. Open the variable explorer and see that a is 1
7. Go to Console 2 (The external one)
8. The variable explorer hasn't changed, even though `a` is not defined

The same problem happens for the plot

### What is the expected output? What do you see instead?
I would expect to see an empty variable explorer (Or plot plugin). Instead I see the data from the previous console I visited.
(Of course the best case would be if the system works with ipkernels) 

## Versions

* Spyder version: dev
* Python version: 3.8.5
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by dalthviz** [comment]
Thanks for the feedback @impact27 !

Pinging @ccordoba12 just in case

**Comment by ccordoba12** [comment]
I agree with @impact27 that this is a very odd UX problem and we need to fix it.
